skbio/alignment/_tabular_msa.py

```diff
@@ -716,8 +730,119 @@ def reindex(self, key=None, keys=None):
             keys_.flags.writeable = False
 
         self._keys = keys_
+        self._minter = minter
+
+    def _fail_if_duplicate_keys(self, keys):
+        # Hashability of keys is implicitly checked here.
+        duplicates = find_duplicates(keys)
+        if duplicates:
+            raise UniqueError(
+                "Keys must be unique. Duplicate keys: %r" % duplicates)
 
     @experimental(as_of='0.4.0-dev')
+    def append(self, sequence, key=None, minter=None):
```

Comment:
Can you swap the order of `key` and `minter` in the signature? (for consistency with `reindex` and constructor)

Also `key` is missing its parameter docstring